[pytorch] Add binding to get nccl version suffix

wesbland opened the pull request:
Summary: Adds a Python to C binding to get the NCCL_SUFFIX value for more accurate NCCL version information and add that to the NCCL version tuple.

Differential Revision: D50978181

wesbland commented:
@pytorchbot label "topic: not user facing"

kwen2501 reviewed (approved):
LGTM.
Waiting to see if CI complains about the change to `def version()`.
If not, we can land it.

wesbland committed:
[pytorch] Add binding to get nccl version suffix (#112884)

Summary:

Adds a Python to C binding to get the NCCL_SUFFIX value for more accurate NCCL version information and add that to the NCCL version tuple.

Reviewed By: kingchc, kwen2501

Differential Revision: D50978181
